### PR TITLE
KMM [Domain] - Add a repository to fetch home information

### PR DESCRIPTION
**Pull request by kibettheophilus**
# Scope

_Please make sure to read the [Contribution Guidelines](https://github.com/droidconKE/droidconKE2023Android/blob/main/CONTRIBUTING.md)
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [ ] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [ ] I have tested the changes added on a physical device
- [ ] I have run `./codeAnalysis.sh` on linux/unix or `codeAnalysys.bat` on windows to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
#292 

## Other testing QA Notes
N/A

Please add a screenshot (if necessary)

### CODE REVIEW

File: shared/domain/src/commonMain/kotlin/com/android254/shared/domain/repos/HomeRepo.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android254.shared.domain.repos
+
+import com.android254.shared.domain.models.HomeDetails
+import kotlinx.coroutines.flow.Flow
+
+interface HomeRepo {
+    fun fetchHomeDetails(): Flow<HomeDetails>
```
Comment (chepsi) [inline shared/domain/src/commonMain/kotlin/com/android254/shared/domain/repos/HomeRepo.kt:22]: Do you mind adding the data implementation with a mock implementation.

Comment (kibettheophilus) [reply shared/domain/src/commonMain/kotlin/com/android254/shared/domain/repos/HomeRepo.kt:22]: No, let me add it... I presume that is in the data module,right?

Comment (kibettheophilus) [reply shared/domain/src/commonMain/kotlin/com/android254/shared/domain/repos/HomeRepo.kt:22]: @chepsi check if I added as expected

File: shared/data/build.gradle.kts [hunk 2]
```diff
@@ -6,8 +21,9 @@ kotlin {
     sourceSets {
         val commonMain by getting {
             dependencies {
-                implementation(project(":shared:datasource"))
-                implementation(project(":shared:domain"))
+                implementation(projects.shared.datasource)
+                implementation(projects.shared.domain)
+                implementation(libs.kotlinx.coroutines.core)
```
Comment (chepsi) [inline shared/data/build.gradle.kts:26]: 🏅

File: shared/data/src/commonMain/kotlin/com/android254/shared/data/repos/HomeRepoImpl.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * limitations under the License.
+ */
+package com.android254.shared.data.repos
+
+import com.android254.shared.domain.models.HomeDetails
+import com.android254.shared.domain.repos.HomeRepo
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+
+class HomeRepoImpl : HomeRepo {
```
Comment (chepsi) [inline shared/data/src/commonMain/kotlin/com/android254/shared/data/repos/HomeRepoImpl.kt:23]: `HomeRepoImpl` -> `HomeDataRepository`
`HomeRepo` -> `HomeRepository`

File: shared/domain/src/commonMain/kotlin/com/android254/shared/domain/models/OrganizingPartners.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+package com.android254.shared.domain.models
+
+data class OrganizingPartners(
+    val organizerName: String,
+    val organizerLogoUrl: String
+)
+
+data class Sponsors(
```
Comment (chepsi) [inline shared/domain/src/commonMain/kotlin/com/android254/shared/domain/models/OrganizingPartners.kt:23]: Add the layer suffix. ie. `SponsorsDomainModel`

Comment (chepsi) [reply shared/domain/src/commonMain/kotlin/com/android254/shared/domain/models/OrganizingPartners.kt:23]: Goes for all the models.

**Review by chepsi — APPROVED** [review]
Looks good. Added a few comments.

**Pull request merged**